effect.cpp

```diff
@@ -435,6 +435,41 @@ string Effects::AssembledMinion::getDescription(const ContentFactory* f) const {
   return "Can be assembled to a " + getName(f);
 }
 
+string Effects::AddAutomatonParts::getPartsNames(const ContentFactory* f) const {
+  string ret = "";
+  bool first = true;
+
+  for(ItemType part_type : part_types) {
+    if(!first) {
+      ret += ", ";
+    }
+    ret += part_type.get(f)->getName();
+    first = false;
+  }
+  return ret;
+}
+
+bool Effects::AddAutomatonParts::applyToCreature(Creature* c, Creature* attacker) const {
+  CHECK(part_types.size() > 0);
+
+  if(c->getSpareAutomatonSlots() < part_types.size()) {
+    return false;
+  }
+
+  for(ItemType part_type : part_types) {
```

Comment:
use "auto&" instead of ItemType. Space between for, if and (. Also in other places.